packages/thirdweb/src/pay/buyWithCrypto/actions/getQuote.ts

```diff
@@ -0,0 +1,316 @@
+import type { Hash } from "viem";
+import { defineChain } from "../../../chains/utils.js";
+import type { ThirdwebClient } from "../../../client/client.js";
+import { getContract } from "../../../contract/contract.js";
+import {
+  approve,
+  type ApproveParams,
+} from "../../../extensions/erc20/write/approve.js";
+import type { PrepareTransactionOptions } from "../../../transaction/prepare-transaction.js";
+import type { BaseTransactionOptions } from "../../../transaction/types.js";
+import { getClientFetch } from "../../../utils/fetch.js";
+import { getPayQuoteEndpoint } from "../utils/definitions.js";
+
+// TODO: add JSDoc description for all properties
+
+/**
+ * The parameters for [`getBuyWithCryptoQuote`](https://portal.thirdweb.com/references/typescript/v5/getBuyWithCryptoQuote) function
+ * It includes information about which tokens to swap, the amount of tokens to swap, slippage, etc.
+ */
+export type getBuyWithCryptoQuoteParams = {
```

Comment:
types are *always* PascalCase
```suggestion
export type GetBuyWithCryptoQuoteParams = {
```